File: Structural/Composite/composite.py
from abc import ABC, abstractmethod
from typing import List

class Component(ABC):
    def __init__(self, name: str):
        self.name = name

    @abstractmethod
    def operation(self) -> None:
        pass

    def add(self, component: 'Component') -> None:
        raise NotImplementedError

    def remove(self, component: 'Component') -> None:
        raise NotImplementedError

    def get_child(self, index: int) -> 'Component':
        raise NotImplementedError

class Leaf(Component):
    def operation(self) -> None:
        print(f"Leaf {self.name} operation")

class Composite(Component):
    def __init__(self, name: str):
        super().__init__(name)
        self.children: List[Component] = []

    def operation(self) -> None:
        print(f"Composite {self.name} operation")
        for child in self.children:
            child.operation()

    def add(self, component: Component) -> None:
        self.children.append(component)

    def remove(self, component: Component) -> None:
        self.children.remove(component)

    def get_child(self, index: int) -> Component:
        return self.children[index]

File: Structural/Composite/composite.cs
public abstract class Component {
    protected string name;
    public Component(string name) {
        this.name = name;
    }
    public abstract void Operation();
    public virtual void Add(Component component) {
        throw new NotSupportedException();
    }
    public virtual void Remove(Component component) {
        throw new NotSupportedException();
    }
    public virtual Component GetChild(int index) {
        throw new NotSupportedException();
    }
}

public class Leaf : Component {
    public Leaf(string name) : base(name) {}
    public override void Operation() {
        Console.WriteLine($"Leaf {name} operation");
    }
}

public class Composite : Component {
    private List<Component> children = new List<Component>();
    public Composite(string name) : base(name) {}
    public override void Operation() {
        Console.WriteLine($"Composite {name} operation");
        foreach (var child in children) {
            child.Operation();
        }
    }
    public override void Add(Component component) {
        children.Add(component);
    }
    public override void Remove(Component component) {
        children.Remove(component);
    }
    public override Component GetChild(int index) {
        return children[index];
    }
}